osf/management/commands/sync_datacite_doi_metadata.py

```diff
@@ -10,31 +10,38 @@
 
 logger = logging.getLogger(__name__)
 
-def sync_datacite_doi_metadata(dry_run=True, retries=4):
+def retry(func, retries=4):
+    for i in range(retries):
+        try:
+            time.sleep(0.3)
+            return func('doi')
+        except DataCiteForbiddenError as e:
+            if i < 1:
```

Comment:
`i` increments here and will start at 0, so this will always raise an exception on the first failure instead of if the number of retries has been exceeded.

Options would be to use `for attempts_remaining in reversed(range(retries))`. Or, since you're already short-circuiting the loop in all cases anyway, to use a `while` loop.

```
attempts_remaining = retries
while True:
  try:
    datacite_response = registration.request_identifier('doi')
  except DataCiteForbiddenError as e:
    if not attempts_remaining:
      raise e
    attempts_remaining -= 1
    time.sleep(10)
  else:
    return datacite_response['doi']
```